keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/Optimizer.java

```diff
@@ -16,17 +16,48 @@
 import io.improbable.keanu.vertices.Vertex;
 
 public interface Optimizer {
-
+    /**
+     * Adds a callback to be called whenever the optimizer evaluates the fitness of a point. E.g. for logging.
+     * @param fitnessCalculationHandler a function to be called whenever the optimizer evaluates the fitness of a point.
+     *                                  The double[] argument to the handler represents the point being evaluated.
+     *                                  The Double argument to the handler represents the fitness of that point.
+     */
     void addFitnessCalculationHandler(BiConsumer<double[], Double> fitnessCalculationHandler);
 
+    /**
+     * Removes a callback function that previously would have been called whenever the optimizer
+     * evaluated the fitness of a point. If the callback is not registered then this function will do nothing.
+     * @param fitnessCalculationHandler the function to be removed from the list of fitness evaluation callbacks
+     */
     void removeFitnessCalculationHandler(BiConsumer<double[], Double> fitnessCalculationHandler);
 
+    /**
+     * This and will use MAP estimation to optimize the values of latent vertices such that the
+     * probability of the whole Bayesian network is maximised.
+     * This method will modify in place the Bayesian network that was used to create this object.
+     *
+     * @return the natural logarithm of the Maximum a posteriori (MAP)
+     */
     double maxAPosteriori();
 
+    /**
+     * This method will use Maximum Likelihood estimation to optimize the values of latent vertices such that
+     * the probability of the whole Bayesian network is maximised.
+     * This method will modify in place the Bayesian network that was used to create this object.
+     *
+     * @return the natural logarithm of the maximum likelihood (MLE)
+     */
     double maxLikelihood();
 
     BayesianNetwork getBayesianNetwork();
 
+    /**
+     * Creates an {@link Optimizer} which optimizes the values of latent variables of the Bayesian network to maximise
```

Comment:
I wonder if it would be clearer to say something like `Creates an Optimizer that provides methods for optimizing the values of latent variables to maximise probability.` I find the use of `overall probability` a little confusing given that ML doesn't take prior probability into account.